graph/db/benchmark_test.go

```diff
@@ -682,6 +684,33 @@ func BenchmarkGraphReadMethods(b *testing.B) {
 				require.NoError(b, err)
 			},
 		},
+		{
+			name: "ForEachNodeCacheable",
+			fn: func(b testing.TB, store V1Store) {
+				err := store.ForEachNodeCacheable(
+					ctx, func(_ route.Vertex,
+						_ *lnwire.FeatureVector) error {
+
+						return nil
```

Comment:
Just a random thought I had when looking at this: Sometimes the Go compiler is very good at optimizing things away when it realizes something doesn't actually do anything (e.g. an empty loop). Probably not the case here, since there are several layers of function calls the empty callback is passed through, so it probably won't be able to optimize away anything. But _just in case_ the compiler ever gets too smart for this, should we just increase a counter in each of these empty callbacks to prevent them from ever being optimized away?

Reply:
cool yes good idea!